content/applications/sales/sales/invoicing/milestone.rst

```diff
@@ -2,32 +2,243 @@
 
-You have to set the product type as *Service* under general
-information and select *Milestones* in the sales tab.
+.. image:: milestone/project-task-option-project-workspace-fields.png
+   :align: center
+   :alt: The Project template and workspace template fields that appear on milestone product.
 
-.. image:: milestone/milestone01.png
-    :align: center
+The :guilabel:`Project Template` field provides template options to use for the project that will be
+created when this specific product is ordered.
+
+The :guilabel:`Workspace Template` field provides template options to use for the workspace that
+will be automatically generated for the project when this specific product is ordered.
+
+.. tip::
+   For organizational purposes, click the :guilabel:`Sales` tab on the product form, and enter a
+   custom 'Milestone' related descriptor in the :guilabel:`Sales Description` field. This
+   information appears in the :guilabel:`Description` column on the :guilabel:`Order Lines` tab of
+   the sales order.
+
+   Or, directly edit/modify the :guilabel:`Description` field on the :guilabel:`Order Lines` tab of
+   the sales order.
+
+   This is *not* a requirement.
 
 Invoice milestones
 ==================
 
-From the sales order, you can manually edit the quantity delivered as
-you complete a milestone.
+.. note::
+   The following flow features a trio of milestone products that have :guilabel:`Service` set as
+   their :guilabel:`Product Type`, and :guilabel:`Task` set on their :guilabel:`Create on Order`
+   field.
+
+    .. image:: milestone/settings-for-workflow.png
+       :align: center
+       :alt: Product with Service "Product Type" and "Task" in the Create on Order field on form.
+
+    Those tasks are then attached to a pre-existing :guilabel:`Project`, which, in this case, is
+    titled, :guilabel:`Rebranding Projects`.
+
+To invoice milestones, create a sales order with the milestone product(s). To do that, go to
+:menuselection:`Sales app --> New`. Doing so reveals a blank quotation form.
+
+From this quotation form, add a :guilabel:`Customer`. Then, click :guilabel:`Add a product` in the
+:guilabel:`Order Lines` tab. Next, add the milestone product(s) to the :guilabel:`Order Lines` tab.
+
+Once the corresponding milestone product(s) have been added, click :guilabel:`Confirm` to confirm
+the order, which turns the quotation into a sales order.
+
+When the order is confirmed, because a product that is configured to be invoiced on milestones, new
+smart buttons appear at the top of the sales order based on what was selected in the
+:guilabel:`Create on Order` field on the product form.
+
+.. image:: milestone/confirmed-sales-order-milestone.png
+   :align: center
+   :alt: Confirmed sales order with milestone products in the order lines tab.
+
+From the sales order, click the :guilabel:`Milestones` smart button. Doing so reveals a blank
+:guilabel:`Milestones` page. Click :guilabel:`New` to add milestones.
```

Comment:
When I click on the smart button, the milestone page is not blank, it populated a list milestones from the SO, which I think is how it should work, right? It looks like you can then add others even from different SOs but I don't think the page should be blank?

Reply:
I might have missed a step, since I created these milestone products on an empty/new Project, but after clicking the "Milestone" smart button, the page *was* blank, and not pre-populated with milestones. If there's additional setup for this, maybe it would be helpful for us to include. Do we have another doc that explains how to create milestones for a project, so it'll be populated at this time? It would be useful to link it here (:

Reply:
I think you're right @Felicious 👍 When I was writing this doc, and going through the flow, I was working off an empty/new Project, and the page was, indeed, blank when the Milestone button was hit. So, perhaps JERO used a populated runbot during that review. So, that said, I think I'll revert it back to "blank" instead of "populated" -- so the reader following along, and *also* working off a new Project, won't be confused when they see a blank page.  Thanks for pointing this out! 👍